src/sentry/uptime/consumers/results_consumer.py

```diff
@@ -17,19 +18,38 @@
     ResultsStrategyFactory,
 )
 from sentry.uptime.detectors.ranking import _get_cluster
+from sentry.uptime.detectors.tasks import set_failed_url
 from sentry.uptime.issue_platform import create_issue_platform_occurrence
-from sentry.uptime.models import ProjectUptimeSubscription, UptimeSubscription
+from sentry.uptime.models import (
+    ProjectUptimeSubscription,
+    ProjectUptimeSubscriptionMode,
+    UptimeSubscription,
+)
+from sentry.uptime.subscriptions.subscriptions import (
+    create_uptime_subscription,
+    delete_project_uptime_subscription,
+    remove_uptime_subscription_if_unused,
+)
 from sentry.utils import metrics
 from sentry.utils.hashlib import md5_text
 
 logger = logging.getLogger(__name__)
 LAST_UPDATE_REDIS_TTL = timedelta(days=7)
+ONBOARDING_FAILURE_THRESHOLD = 3
+ONBOARDING_FAILURE_REDIS_TTL = timedelta(days=2)
+ONBOARDING_MONITOR_PERIOD = timedelta(days=3)
+# How frequently we should run active auto-detected subscriptions
+AUTO_DETECTED_ACTIVE_SUBSCRIPTION_INTERVAL_SECONDS = int(timedelta(minutes=5).total_seconds())
```

Comment:
Can we add some comments?

```suggestion
ONBOARDING_FAILURE_THRESHOLD = 3
"""
When onboarding a new subscription how many TOTAL failures are allowed to happen during
the ONBOARDING_MONITOR_PERIOD before we consider the subscription to have failed onboarding.
"""

ONBOARDING_MONITOR_PERIOD = timedelta(days=3)
"""
How long does a subscription need to be in the `ProjectUptimeSubscriptionMode.AUTO_DETECTED_ONBOARDING` 
mode before it can graduate to `ProjectUptimeSubscriptionMode.AUTO_DETECTED_ACTIVE`?
During this time the subscription must not fail a check more than `ONBOARDING_FAILURE_THRESHOLD` times.
"""

# The TTL of the redis key used to track the failure counts for a subscription
# In `ProjectUptimeSubscriptionMode.AUTO_DETECTED_ONBOARDING` mode. Must be longer
# Than the ONBOARDING_MONITOR_PERIOD.
ONBOARDING_FAILURE_REDIS_TTL = ONBOARDING_MONITOR_PERIOD + timedelta(days=1)

AUTO_DETECTED_ACTIVE_SUBSCRIPTION_INTERVAL_SECONDS = int(timedelta(minutes=5).total_seconds())
"""
How frequently are active auto-detected subscriptions configured to run.
""" 
```

(feel free if you want to use normal comments, I just like the docstring comments since pyright shows them as editor hints on hover)